src/hooks/useTokenIdentityRegistryAgents.ts

```diff
@@ -0,0 +1,76 @@
+import {
+  readIdentityRegistryIsAgent,
+  watchIdentityRegistryAgentAddedEvent,
+} from "@/services/contracts/wagmiGenActions";
+import { EvmAddress } from "@/types/types";
+import { useQueries } from "@tanstack/react-query";
+import { useEffect, useState } from "react";
+import { WatchContractEventReturnType } from "viem";
+import { QueryKeys } from "./types";
+
+export function useTokenIdentityRegistryAgents(registry?: EvmAddress) {
+  const [agents, setAgents] = useState<Array<string>>([]);
+  const [uniqueAgents, setUniqueAgents] = useState<Array<string>>([]);
```

Comment:
can we filter out agents in `onLogs` right away? if we don't need to use unfiltered agents